component/loki/source/kubernetes/kubetail/tailer.go

```diff
@@ -167,9 +169,51 @@ func (t *tailer) tail(ctx context.Context, handler loki.EntryHandler) error {
 	if err != nil {
 		return err
 	}
+
+	// Create a new rolling average calculator to determine the average delta
+	// time between log entries.
+	//
+	// Here, we track the most recent 10,000 delta times to compute a fairly
+	// accurate average. If there are less than 100 deltas stored, the average
+	// time defaults to 1h.
+	//
+	// The computed average will never be less than the minimum of 2s.
+	calc := newRollingAverageCalculator(10000, 100, 2*time.Second, maxTailerLifetime)
```

Comment:
Does that mean the max refresh interval is 6 seconds? I think we won't thrash frequent updates since this makes a new tailer each time, and that takes time to reach stable averages?

Reply:
No, the _minimum_ refresh interval is 6 seconds; the default maximum is 3h, and with the way this is written it's not possible for that to get any higher. 

It's not possible to go above 3h because you need at least 100 log lines for the average to be calculated; logging less frequently than once every 3 hours means the tailer would've already been reset (including resetting the calculator) before the true average was detected.